src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/pages/edit/Edit.tsx

```diff
@@ -124,10 +134,28 @@ const Edit = () => {
     }
   }, [measurementFamily, locale, saveMeasurementFamily, notify, __, setErrors]);
 
-  const handleNewUnit = useCallback(
-    (unit: Unit) => {
-      null !== measurementFamily && setMeasurementFamily(addUnit(measurementFamily, unit));
-    },
+  const removeMeasurementFamily = useMeasurementFamilyRemover();
+  const removeMeasurementFamilyHandler = useCallback(async () => {
+    try {
+      const response = await removeMeasurementFamily(measurementFamilyCode);
+
+      switch (response) {
+        case MeasurementFamilyRemoverResult.Success:
+          notify(NotificationLevel.SUCCESS, __('measurements.family.delete.flash.success'));
+          history.push('/');
+          break;
+        case MeasurementFamilyRemoverResult.NotFound:
+        case MeasurementFamilyRemoverResult.Unprocessable:
+          break;
+      }
+    } catch (error) {
+      console.error(error);
+      notify(NotificationLevel.ERROR, __('measurements.family.delete.flash.error'));
+    }
+  }, [measurementFamilyCode, removeMeasurementFamily, history]);
+
+  const newUnitHandler = useCallback(
+    (unit: Unit) => null !== measurementFamily && setMeasurementFamily(addUnit(measurementFamily, unit)),
     [setMeasurementFamily, measurementFamily]
   );
```

Comment:
```suggestion
  const newUnitHandler = useCallback((unit: Unit) => {
    if (null === measurementFamily) {
      return;
    }

    setMeasurementFamily(addUnit(measurementFamily, unit))
  }, [setMeasurementFamily, measurementFamily]);
```

just a suggestion, to say consistent with the other handler defined there.